c10/xpu/test/impl/XPUCachingAllocatorTest.cpp

```diff
@@ -75,6 +76,46 @@ TEST(XPUCachingAllocatorTest, AllocateMemory) {
   for (const auto i : c10::irange(numel)) {
     EXPECT_EQ(hostData[i], i);
   }
+  c10::xpu::XPUCachingAllocator::emptyCache();
+}
+
+TEST(XPUCachingAllocatorTest, DeviceCachingAllocateByExternalStream) {
```

Comment:
Can we add a case to check the scenario that pointers to sycl::queue and its copies.
Assert the returned ExternalStream-s are totally different.

Reply:
Added.